src/session/session_compact.c

```diff
@@ -360,6 +360,31 @@ __compact_worker(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __wt_compact_uri_check --
+ *     Function to check whether the specified URI can be compacted.
+ */
+int
+__wt_compact_uri_check(WT_SESSION_IMPL *session, const char *uri)
```

Comment:
I started with this implementation but `__wt_session_compact` contains additional check that could be included:

```
    /* Disallow objects in the WiredTiger name space. */
    WT_ERR(__wt_str_name_check(session, uri));
    if (!WT_PREFIX_MATCH(uri, "colgroup:") && !WT_PREFIX_MATCH(uri, "file:") &&
      !WT_PREFIX_MATCH(uri, "index:") && !WT_PREFIX_MATCH(uri, "lsm:") &&
      !WT_PREFIX_MATCH(uri, "table:") && !WT_PREFIX_MATCH(uri, "tiered:")) {
        if ((dsrc = __wt_schema_get_source(session, uri)) != NULL)
            ret = dsrc->compact == NULL ?
              __wt_object_unsupported(session, uri) :
              dsrc->compact(dsrc, wt_session, uri, (WT_CONFIG_ARG *)cfg);
        else
            ret = __wt_bad_object_type(session, uri);
        goto err;
    }
```

What is bothering me is the call to `dsrc->compact(dsrc, wt_session, uri, (WT_CONFIG_ARG *)cfg)` inside those checks.